src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { ReactComponent as CheckmarkNormal } from './CheckmarkNormal.svg';
+import { ReactComponent as CheckmarkCompact } from './CheckmarkCompact.svg';
+import classes from './Checkbox.module.css';
+
+export interface CheckboxProps {
+  checkboxId?: string;
+  checked?: boolean;
+  className?: string;
+  compact?: boolean;
+  disabled?: boolean;
+  error?: boolean;
+  label?: string;
+  name?: string;
+  readOnly?: boolean;
+}
+
+export const Checkbox = ({
+  checkboxId,
+  checked,
+  className,
+  compact,
+  disabled,
+  error,
+  label,
+  name,
+  readOnly,
+}: CheckboxProps) => {
+  const [isChecked, setIsChecked] = useState<boolean>(!!checked);
+
+  useEffect(() => setIsChecked(!!checked), [checked]);
+
+  const inputId =
+    checkboxId ||
+    'checkbox-' + window.crypto.getRandomValues(new Uint32Array(1));
```

Comment:
Should we just require consumers to supply an ID instead?

If we want to keep it optional, I think its better to use the [`useId` hook](https://reactjs.org/docs/hooks-reference.html#useid) instead.

We should probably also just stick with `id` instead of `checkboxId` as the prop name to keep naming consistent with the native HTML naming.

```suggestion
  const inputId = id || useId();
```

Reply:
I made it required simply to make life easier for the consumers, but it doesn't really matter to me. I can make it required if you think that's better. Anyways, I didn't know about the `useId` hook, so thanks for the suggestion!

When it comes to naming, I chose `checkboxId` to make it clear that it's not the id of the root element, but the checkbox inside. I think `id` might be confusing for the consumers if they were to use it for something else, i.e. custom styling, as it usually refers to the id of the component's root element.

Reply:
Might be good to dont have to provide an id in some cases, so leaving it optional is fine 🙂

Yeah, it could perhaps be misinterpreted by using just `id`. Lets stick with `checkboxId` 🙂

We are creating a highly opinionated design system (and component library), so custom styling should be discouraged. Perhaps we should even remove the `classname` prop?